### DISCUSSION TITLE
父路由重定向问题

### DISCUSSION THREAD

**Discussion by erraleio** [umijs/umi#12307] [Q&A]
父路由重定向到子一个固定的子路由，登录不同账号这个子路由会没有权限，怎么让父路由动态重定向的地址是第一个有权限的子路由
 {
    path: '/student',
    name: '学生',
    icon: 'SearchOutlined',
    routes: [
      {
        redirect: 'analysis',
        path: '/student',
      },
      {
        name: '课程分析',
        path: 'analysis',
        component: './student/analysis/index',
        access: 'routerAuthor',
      },
  ]
}

**Comment by fz6m** [comment]
你在要跳 url 之前，得去找一下这个用户第一个有权限的路由是哪个，然后再去跳有权限的路由，可能要写一个通用的跳转方法。

如果想跳之后解决，最好把 access 方案改成运行时的（不配在静态路由表配置里，或者用一个其他字段，然后用 [`useRouteProps`](https://umijs.org/docs/api/api#userouteprops) 取一下当前是哪个 access ），比如搭配 `useAccess` 用 [`<Access></Access>`](https://umijs.org/docs/max/access#access-1) 去包裹，在渲染的时候自己去判断下当前有没有权限，没有的话就跳到有权限的路由去。

或者还有一个更晚的时机，就是自定义下 layout 的无权限 `unAccessible` 展示的组件，如果这个组件展示出来了就跳其他路由。
